splice_ck/src/main/java/com/splicemachine/ck/HBaseInspector.java

```diff
@@ -0,0 +1,176 @@
+                                    tableId.set(new Pair<>(userData.getTimestamp(), er.getColumn(SYSTABLESRowFactory.SYSTABLES_TABLEID).getString()));
+                                }
+                            }
+                        }
+                    };
+                    for (Cell cell : r.listCells()) {
+                        cellVisitor.visit(cell);
+                    }
+                }
+                if (tableId.get() == null) {
+                    throw new TableNotFoundException();
+                }
+                return constructSchemaFromSysColsTable(tableId.get().getSecond());
+            }
+        }
+    }
+
+    private Utils.Tabular constructSchemaFromSysColsTable(String tableId) throws Exception {
+        String systableId = getHbaseRegionOf(SYSCOLUMNSRowFactory.TABLENAME_STRING);
+        try (Connection conn = ConnectionFactory.createConnection(config)) {
+            Table table = conn.getTable(TableName.valueOf(systableId));
+            Scan scan = new Scan().addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.PACKED_COLUMN_BYTES);
+            try (ResultScanner results = table.getScanner(scan)) {
+                final ValContainer<Utils.Tabular> schema = new ValContainer<>(new Utils.Tabular("column index", "column name", "column type"));
+                for (Result r : results) {
+                    ICellVisitor cellVisitor = new ICellVisitor() {
+                        @Override
+                        protected void visitUserData(Cell userData) throws StandardException {
+                            UserDataDecoder decoder = new SysColsDataDecoder();
+                            ExecRow er = decoder.decode(userData);
+                            if (er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_REFERENCEID) != null
+                                    && tableId.equals(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_REFERENCEID).getString())) {
+                                schema.get().addRow(Integer.toString(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNNUMBER).getInt()),
+                                        CheckNull(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNNAME).toString()),
+                                        CheckNull(((TypeDescriptorImpl)(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNDATATYPE)).getObject()).getTypeName()));
+                            }
+                        }
+                    };
+                    for (Cell cell : r.listCells()) {
+                        cellVisitor.visit(cell);
+                    }
+                }
+                return schema.get();
+            }
+        }
+    }
+
+    public String getHbaseRegionOf(String tableName) throws Exception {
+        Utils.Tabular results = getListTables();
+        Utils.Tabular.Row r = results.rows.stream().filter(result -> result.cols.get(2 /*name*/).equals(tableName)
+                && result.cols.get(3 /*index*/).equals("NULL")).findAny().orElse(null);
```

Comment:
index? isn't that transactionId?

Reply:
Actually yes, it is the `index`, in hbase one could find many tables sharing the same table name (as far as I understand, base table and its indices share the same name), one way to tell which one is the base table is by checking that it doesn't have an index name.